readthedocs/proxito/views/mixins.py

```diff
@@ -323,6 +323,8 @@ def system_redirect(
         log.debug(
             "System Redirect.", host=request.get_host(), from_url=filename, to_url=to
         )
+        # All system redirects can be cached, since the final URL will check for authz.
```

Comment:
Great comment 👍 What about an idea where we could do like `self.cache_response = CacheResponse(cache=True, reason='All system redirects can be cached, since the final URL will check for authz.')` and force a `reason` in our code? I think we always want to be really clear/intentional about caching. 

Similarly, I think we probably want the ability to specify how long to cache things? For example, some things should be cached for a long time, and others probably less time...

Reply:
This could be a good idea to have more control over caching, another way could be to call function instead of defining an attribute, like `self.cache_response(ttl=..., reason=...)`.

> Similarly, I think we probably want the ability to specify how long to cache things? For example, some things should be cached for a long time, and others probably less time...

We are tracking that at https://github.com/readthedocs/readthedocs.org/issues/10035

Reply:
Sounds good.